apps/examples/testcase/le_tc/kernel/tc_libc_timer.c

```diff
@@ -437,6 +450,70 @@ static void tc_libc_timer_strptime(void)
 	sf_tm.tm_year = 2018 - TM_YEAR_BASE;
 
 	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("test", "%%", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail/05/18", "%D", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("24:42", "%R", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail:42:13 PM", "%r", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("12:42:Fail", "%T", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("10:Fail:13", "%X", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("10/Fail/18", "%x", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Jan", "%B", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%B", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%C", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("13 AM", "%H %p", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("12 AM", "%H %p", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+
+	ret = strptime("13 PM", "%H %p", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("11 PM", "%H %p", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+
+	ret = strptime("FM", "%p", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%U", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%Y", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%y", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
```

Comment:
Please check that which case needs memset.

Reply:
modify